app/lib/screens/ChatProfileScreen/RoomLinkSetting.dart

```diff
@@ -0,0 +1,181 @@
+                    style: TextStyle(
+                      color: AppCommonTheme.primaryColor,
+                      fontSize: 16.0,
+                    ),
+                  );
+                } else {
+                  return const Text('Loading Name');
+                }
+              },
+            ),
+            Container(
+              margin: EdgeInsets.only(top: 20.0, bottom: 10.0),
+              child: Text(
+                'Expires After',
+                style: TextStyle(color: Colors.white, fontSize: 18.0),
+              ),
+            ),
+            Flexible(
+              child: GridView.builder(
+                shrinkWrap: true,
+                gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                  crossAxisCount: 5,
+                  crossAxisSpacing: 10,
+                  mainAxisSpacing: 20,
+                ),
+                itemCount: timeArr.length,
+                itemBuilder: (BuildContext content, index) {
+                  return GestureDetector(
+                    onTap: () {
+                      setState(() {
+                        timeIndexing = index;
+                        if (!selectedTimeIndexList.contains(index)) {
+                          selectedTimeIndexList.clear();
+                          selectedTimeIndexList.add(index);
+                        } else {
+                          selectedTimeIndexList.clear();
+                        }
+                      });
+                    },
+                    child: Container(
+                      child: Center(
+                        child: Text(
+                          timeArr[index],
+                          style: TextStyle(color: Colors.white),
+                        ),
+                      ),
+                      decoration: BoxDecoration(
+                        color: selectedTimeIndexList.contains(index)
+                            ? AppCommonTheme.primaryColor
+                            : AppCommonTheme.darkShade,
+                        borderRadius: BorderRadius.circular(15),
+                      ),
+                    ),
+                  );
+                },
+              ),
+            ),
+            Container(
+              margin: EdgeInsets.only(top: 20.0, bottom: 10.0),
+              child: Text(
+                'Max Uses',
+                style: TextStyle(color: Colors.white, fontSize: 18.0),
+              ),
+            ),
+            GridView.builder(
+              shrinkWrap: true,
+              gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
+                crossAxisCount: 6,
+                crossAxisSpacing: 10,
+                mainAxisSpacing: 20,
+              ),
+              itemCount: usesArr.length,
+              itemBuilder: (BuildContext content, index) {
+                return GestureDetector(
+                  onTap: () {
+                    setState(() {
```

Comment:
This is just as a reminder that since we have switched to GetX as our main state handler for flutter UI, we want to use ```setState()``` as minimum as possible. If you think this action can be doable with GetX, please do so.  Note that using ```setState()``` it'll re-render the widget as a whole which we, in some cases don't necessarily want.

Reply:
This only updates the selection list, so I think we can do that by creating function in chatController, which updates index.

Reply:
setState() is just there to show how it is supposed to work incase someone else implements the functionality later on

Reply:
Sure, I just wants to give heads up that we would be approach like this later on. :)